service/src/main/java/org/cbioportal/service/impl/UuidDataAccessTokenServiceImpl.java

```diff
@@ -0,0 +1,203 @@
+
+@Service
+@Component
+public class UuidDataAccessTokenServiceImpl implements DataAccessTokenService {
+
+    @Autowired
+    private DataAccessTokenRepository dataAccessTokenRepository;
+
+    @Value("${jwt.ttl_seconds}")
+    private int datTtlSeconds;
+
+    @Value("${max.number.dat}")
+    private int maxNumberOfAccessTokens;
+
+    private static final Log log = LogFactory.getLog(UuidDataAccessTokenServiceImpl.class);
+
+    // create a data access token (randomly generated UUID) and insert corresponding record into table with parts:
+    // username
+    // uuid
+    // expiration date (current time + 1 month)
+    @Override
+    public DataAccessToken createDataAccessToken(String username, boolean allowRevocationOfOtherTokens) {
+        if (username == null || username.trim().length() == 0) {
+            throw new IllegalArgumentException("subject cannot be empty");
+        }
+        String uuid = UUID.randomUUID().toString();
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.SECOND, datTtlSeconds);
+        Date expirationDate = calendar.getTime();
+       
+        DataAccessToken dataAccessToken = new DataAccessToken(); 
+        dataAccessToken.setToken(uuid);
+        dataAccessToken.setUsername(username);
+        dataAccessToken.setExpiration(expirationDate);
+        if (getNumberOfTokensForUsername(username) == maxNumberOfAccessTokens) {
+            revokeOldestDataAccessTokenForUsername(username);
+        }
+        dataAccessTokenRepository.addDataAccessToken(dataAccessToken);
+        return dataAccessToken;
+    }
+
+    // get all user tokens/uuids
+    @Override
+    public List<DataAccessToken> getAllDataAccessTokens(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        return allDataAccessTokens;
+    }
+    
+    // get data access token for a given username
+    // get all data access tokens available for user in table and return newest one (uuid)
+    @Override
+    public DataAccessToken getDataAccessToken(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        DataAccessToken newestDataAccessToken = getNewestDataAccessToken(allDataAccessTokens);
+        return newestDataAccessToken;
+    }
+
+    @Override
+    public DataAccessToken getDataAccessTokenInfo(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        return dataAccessToken;
+    }
+
+    @Override
+    public void revokeAllDataAccessTokens(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        for (DataAccessToken dataAccessToken : allDataAccessTokens) {
+            dataAccessTokenRepository.removeDataAccessToken(dataAccessToken.getToken());
```

Comment:
we can avoid this loop by adding the "removeAllDataAccessTokensForUser" mybatis mapper, which can do it with one step